pkg/util/gvk-gvr-mapper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GvkGvrMapper is a thread-safe mapping between GVKs and GVRs.
+type GvkGvrMapper interface {
+	// Add adds a mapping between the given GVK and GVR.
+	Add(gvk schema.GroupVersionKind, gvr schema.GroupVersionResource)
+	// DeleteByGvkKey deletes the mapping associated with the given GVK.
+	// The key format is that outputted by utils.KeyForGroupVersionKind.
+	DeleteByGvkKey(string)
+	// DeleteByGvrKey deletes the mapping associated with the given GVR.
+	// The key format is that outputted by utils.KeyForGroupVersionResource.
+	DeleteByGvrKey(string)
+	// GetGvr returns the GVR associated with the given GVK if it exists.
+	// The returned boolean indicates whether it exists or not.
+	GetGvr(gvk schema.GroupVersionKind) (*schema.GroupVersionResource, bool)
```

Comment:
Why return a _pointer to_ the GVR instead of the GVR itself? It seems to me that returning the pointer just adds work.